### PR TITLE
Restore event messages for replica sets in the deployment controller

### PR DESCRIPTION
**Pull request by 0xmichalis**
Needed to unblock release upgrade tests (see https://github.com/kubernetes/kubernetes/issues/36453)

@kubernetes/deployment ptal

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36579)

### CODE REVIEW

**Comment by kargakis** [comment]
@krousey fyi

File: test/e2e/deployment.go [hunk 1]
```diff
@@ -398,8 +398,8 @@ func testRollingUpdateDeploymentEvents(f *framework.Framework) {
 	newRS, err := deploymentutil.GetNewReplicaSet(deployment, c)
 	Expect(err).NotTo(HaveOccurred())
 	Expect(newRS).NotTo(Equal(nil))
-	Expect(events.Items[0].Message).Should(Equal(fmt.Sprintf("Created new replica set %q and scaled up to 1", newRS.Name)))
-	Expect(events.Items[1].Message).Should(Equal(fmt.Sprintf("Scaled down replica set %q to 0", rsName)))
+	Expect(events.Items[0].Message).Should(Equal(fmt.Sprintf("Scaled up replica set %s to 1", newRS.Name)))
```
Comment (mfojtik) [inline test/e2e/deployment.go]: @kargakis do we want to rely on these events or is there a better way to verify?

**Comment by kargakis** [comment]
@k8s-bot unit test this https://github.com/kubernetes/kubernetes/issues/36473

**Comment by kargakis** [comment]
@k8s-bot unit test this #36473

**Comment by kargakis** [comment]
@janetkuo can you ptal?

**Comment by janetkuo** [comment]
Probably need a `TODO`, but it's optional

**Comment by janetkuo** [comment]
@k8s-bot unit test this

**Pull request merged**